### PR TITLE
shm: writeback support

### PR DESCRIPTION
**Pull request by Drakulix**
Draft PR, because I still need to test the new functionality in a compositor, but ready for review and anvil still runs fine!

### CODE REVIEW

File: src/wayland/shm/mod.rs [hunk 1]
```diff
@@ -219,6 +219,35 @@ where
...
+/// If the buffer is managed by the provided `ShmGlobal`, its contents are
+/// extracted and the closure is extracted with them:
+///
+/// - The first argument is a data slice of the contents of the pool
+/// - The second argument is the specification of this buffer is this pool
+///
+/// If the buffer is not managed by the provided `ShmGlobal`, the closure is not called
+/// and this method will return `Err(BufferAccessError::NotManaged)` (this will be the case for an
+/// EGL buffer for example).
+pub fn with_buffer_contents_mut<F, T>(buffer: &wl_buffer::WlBuffer, f: F) -> Result<T, BufferAccessError>
```
Comment (i509VCB) [inline src/wayland/shm/mod.rs:233]: So we want to guard against the future access flags in wl_shm V3? Not every error will end up being NotManaged in the future

Comment (Drakulix) [reply src/wayland/shm/mod.rs:233]: Do you have some reference, what might happen in the future?

Comment (i509VCB) [reply src/wayland/shm/mod.rs:233]: https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/161/ is the proposal

eli5: In version 3, a new request is added which allows specifying what the server can do with the buffer (read and or write). For v2 and v1, assume the buffer is always readable and writable.

I believe there are some semantics changing related to resizing, but those are out to scope for this pull request.

Comment (Drakulix) [reply src/wayland/shm/mod.rs:233]: Ok, so we want a `not-writable` error type, got it.

Comment (Drakulix) [reply src/wayland/shm/mod.rs:241]: Added new Error-variants for non-readable or non-writable buffers.

**Review by i509VCB — APPROVED** [review]
Looks good, actually wiring up the permissions for read/write flags is probably for a different day.
